File: src/main/kotlin/ru/blimfy/controllers/dto/SignUpDto.kt
package ru.blimfy.controllers.dto

/**
 * DTO для регистрации нового пользователя.
 *
 * @property email электронная почта.
 * @property username имя пользователя.
 * @property password пароль.
 * @author Владислав Кузнецов.
 * @since 0.0.1.
 */
data class SignUpDto(val email: String, val username: String, val password: String)

File: ui/src/api/dto/request/auth/SignUpDto.ts
/**
 * DTO с информацией о новом пользователе.
 *
 * @author Владислав Кузнецов.
 * @since 0.0.1.
 */
export class SignUpDto {
  /**
   * Электронная почта нового пользователя.
   */
  email: string = ""

  /**
   * Логин нового пользователя.
   */
  username: string = ""

  /**
   * Пароль нового пользователя.
   */
  password: string = ""

  /**
   * День рождения пользователя.
   */
  day: number | undefined = undefined

  /**
   * Месяц рождения пользователя.
   */
  month: number | undefined = undefined

  /**
   * Год рождения пользователя.
   */
  year: number | undefined = undefined
}
